api/adduploadedfile.go

```diff
@@ -0,0 +1,70 @@
+package api
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/ONSdigital/dp-import-api/models"
+	"github.com/ONSdigital/go-ns/audit"
+	"github.com/ONSdigital/go-ns/common"
+	"github.com/ONSdigital/go-ns/log"
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+)
+
+func (api *ImportAPI) addUploadedFileHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	vars := mux.Vars(r)
+	jobID := vars["id"]
+	logData := log.Data{jobIDKey: jobID}
+	auditParams := common.Params{jobIDKey: jobID}
+
+	// record attempt to add uploaded file to job
+	if auditError := api.auditor.Record(ctx, uploadFileAction, audit.Attempted, auditParams); auditError != nil {
+		handleErr(ctx, w, auditError, logData)
+		return
+	}
+
+	uploadedFile, err := models.CreateUploadedFile(r.Body)
+	defer r.Body.Close()
+	if err != nil {
+		log.ErrorCtx(ctx, errors.WithMessage(err, "addUploadFile endpoint: failed to create uploaded file resource"), logData)
+
+		// record failure to add uploaded file
+		if auditError := api.auditor.Record(ctx, uploadFileAction, audit.Unsuccessful, auditParams); auditError != nil {
+			err = auditError
+		}
+
+		handleErr(ctx, w, err, nil)
+		return
+	}
+
+	logData["file"] = uploadedFile
+	auditParams["fileAlias"] = uploadedFile.AliasName
+	auditParams["fileURL"] = uploadedFile.URL
+
+	if err := api.addUploadFile(ctx, uploadedFile, jobID, auditParams, logData); err != nil {
+		// record unsuccessful attempt to add uploaded file to job
+		if auditError := api.auditor.Record(ctx, uploadFileAction, audit.Unsuccessful, auditParams); auditError != nil {
+			err = auditError
+		}
+
+		handleErr(ctx, w, err, logData)
+		return
+	}
+
+	// record successful attempt to add uploaded file to job
+	api.auditor.Record(ctx, uploadFileAction, audit.Successful, auditParams)
```

Comment:
You aren't handling the error

Reply:
So the new auditor will log an audit error if it fails to audit the action success. The error that gets returned is just informative for other scenarios but here we do not act on it as the user has successfully added the uploaded file details to the job resource and hence a 200 response (NOT a 500) needs to be returned.